File: app/src/main/java/com/example/practice/adapter/SystemNavigationAdapter.java
package com.example.practice.adapter;

import android.view.ViewGroup;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * Created By 大苏打
 * on 2020/10/23
 */
public class SystemNavigationAdapter extends FragmentPagerAdapter{
    private List<Fragment>list;
    private String []title;

    public SystemNavigationAdapter(@NonNull FragmentManager fm, List<Fragment> list, String[] title){
        super(fm);
        this.list = list;
        this.title = title;
    }

    @NonNull
    @Override
    public Fragment getItem(int position){
        return list.get(position);
    }

    @Override
    public int getCount(){
        return list.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position){
        return title[position];
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object){
        super.destroyItem(container, position, object);
    }
}

File: app/src/main/java/com/example/practice/adapter/SystemNavigationAdapter.kt
package com.example.practice.adapter

import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

/**
 * Created By 大苏打
 * on 2020/10/23
 */
class SystemNavigationAdapter(
    fm: FragmentManager,
    private val list: List<Fragment>,
    private val title: MutableList<String>
) : FragmentPagerAdapter(fm) {
    override fun getItem(position: Int): Fragment {
        return list[position]
    }

    override fun getCount(): Int {
        return list.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return title[position]
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        super.destroyItem(container, position, `object`)
    }
}